drivers/pktmem.c

```diff
@@ -17,15 +17,24 @@
 #include <string.h>
 
 
+// FIXME: transfer to appropriate file with platform-specific definitions
+#if defined(__ARM_ARCH_7EM__)
+#define CACHE_LINE_SIZE 32
+#else
 #define CACHE_LINE_SIZE 64
-#define PAGE_SIZE       4096
+#endif
 
 #define TRUE_LRU 0
 
+#define PKT_BUF_SIZE (2048 - CACHE_LINE_SIZE)
```

Comment:
- what about `imx-enet.c:151` (`state->mmio->MRBR = ENET_BUFFER_SIZE;  // FIXME: coerce with net_allocPktBuf()`)
- incomplete solution, munmap is called with `PAGE_SIZE` as an argument
- is 2048 - 32 really needed as a packet buf? (just asking whether we could use eg.`3 * 512 - 32`)
- IMHO better solution would be to define eg. `PKT_BUF_ALLOC_SIZE` here (depending on `PAGE_SIZE`) and use it in implementation (instead of ifdefed code)
- please note that `mmap` works on full pages, so it's best to alloc multiplicity of PAGE_SIZE (otherwise remaining portion of the last page will never be used)

Reply:
1. should I extract the `PKT_BUF_SIZE` to `drivers/pktmem.h` and get this value in `drivers/imx-enet.c` from there?
2. good point
3. `3 * 512 - 32` < 1518, which is the max ethernet frame size, though it's just enough without the optimisation
4. yes

Reply:
- IMHO for small targets it's better not to waste memory over minuscule performance improvement due to cache usage
- putting `PKT_BUF_SIZE` in header is a bad idea as the value is not const (it's platform-dependant, actual computation method is implementation detail and it should be hidden from external API). Maybe provide function returning the buffer size instead?